crossbeam-epoch/src/epoch.rs

```diff
@@ -102,13 +102,31 @@ impl AtomicEpoch {
 
     /// Stores a value into the atomic epoch if the current value is the same as `current`.
     ///
-    /// The return value is always the previous value. If it is equal to `current`, then the value
-    /// is updated.
+    /// The return value is a result indicating whether the new value was written and containing
+    /// the previous value. On success this value is guaranteed to be equal to `current`.
     ///
-    /// The `Ordering` argument describes the memory ordering of this operation.
+    /// `compare_exchange` takes two `Ordering` arguments to describe the memory
+    /// ordering of this operation. `success` describes the required ordering for the
+    /// read-modify-write operation that takes place if the comparison with `current` succeeds.
+    /// `failure` describes the required ordering for the load operation that takes place when
+    /// the comparison fails. Using `Acquire` as success ordering makes the store part
+    /// of this operation `Relaxed`, and using `Release` makes the successful load
+    /// `Relaxed`. The failure ordering can only be `SeqCst`, `Acquire` or `Relaxed`
+    /// and must be equivalent to or weaker than the success ordering.
     #[inline]
-    pub fn compare_and_swap(&self, current: Epoch, new: Epoch, ord: Ordering) -> Epoch {
-        let data = self.data.compare_and_swap(current.data, new.data, ord);
-        Epoch { data }
+    pub fn compare_exchange(
```

Comment:
Note: This is marked as `pub`, but this is a function used only within the crate and not exposed, so this change does not break the public API.

Reply:
- By "exposed", you mean it's not part of the public API?
- Even if it is public, I vote for chaging like you suggested.  We're before 1.0 :)

Reply:
> * By "exposed", you mean it's not part of the public API?

Yes, this function and `AtomicEpoch` that implements this function is not part of the public API: https://docs.rs/crossbeam-epoch/0.9.1/crossbeam_epoch/#structs